### PR TITLE
add test case for object copy in versioning suspended bucket

### PR DESCRIPTION
**Pull request by jzhu116-bloomberg**
add test case test_versioning_obj_suspended_copy() for https://tracker.ceph.com/issues/64340 

Signed-off-by: Jane Zhu [jzhu116@bloomberg.net](mailto:jzhu116@bloomberg.net)

### CODE REVIEW

**Review by cbodley — APPROVED** [review]
verified that this test fails against ceph main

File: s3tests_boto3/functional/test_s3.py [hunk 1]
```diff
@@ -7509,6 +7509,40 @@ def test_versioning_obj_suspend_versions():
...
+    # copy to another object
+    key2 = 'testobj2'
+    copy_source = {'Bucket': bucket_name, 'Key': key1}
+    client.copy_object(Bucket=bucket_name, Key=key2, CopySource=copy_source)
+
+    # delete the source object. keep the 'null' entry in version_ids
+    client.delete_object(Bucket=bucket_name, Key=key1)
+
+    # get the target object
+    response = client.get_object(Bucket=bucket_name, Key=key2)
```
Comment (cbodley) [inline s3tests_boto3/functional/test_s3.py:7536]: when [this job](http://qa-proxy.ceph.com/teuthology/cbodley-2024-02-11_16:11:11-rgw-wip-cbodley2-testing-distro-default-smithi/7556252/teuthology.log) ran against dbstore in the rgw/dbstore suite, it failed here with:
> An error occurred (NoSuchKey) when calling the GetObject operation: None

not sure what exactly dbstore is doing wrong here, but lets tag this test with `@pytest.mark.fails_on_dbstore` for now

Comment (jzhu116-bloomberg) [reply s3tests_boto3/functional/test_s3.py:7537]: done

**Comment by cbodley** [comment]
cherry-picked to ceph-master

**Pull request merged**